File: 203. Remove Linked List Elements/solution.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func removeElements(head *ListNode, val int) *ListNode {
    var prev *ListNode
    curr := head
    for curr != nil {
        if curr.Val == val {
            if prev == nil {
                head = head.Next
                curr = head
            } else {
                prev.Next = curr.Next
                curr = curr.Next
            }
        } else {
            prev = curr
            curr = curr.Next
        }
    }
    return head
}

File: 203. Remove Linked List Elements/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def removeElements(self, head, val):
        """
        :type head: ListNode
        :type val: int
        :rtype: ListNode
        """
        # edge cases
        if not head:
            return head
        
        # make dummy head
        node = ListNode(None)
        node.next = head
        head = node
        node = node.next
        prev = head
        
        # remove nodes with val 
        while node:
            if node.val == val:
                prev.next = node.next
                node = node.next
            else:
                prev = node
                node = node.next
                
        return head.next
        
